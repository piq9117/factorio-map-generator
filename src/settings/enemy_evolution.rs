use serde::ser::SerializeStruct;
use serde::{Serialize, Serializer};
pub struct EnemyEvolutionSettings {
    pub enabled: bool,
    pub time_factor: Option<f32>,
    pub destroy_factor: Option<f32>,
    pub pollution_factor: Option<f32>,
}

impl Serialize for EnemyEvolutionSettings {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: Serializer,
    {
        let mut s = serializer.serialize_struct("EnemyEvolutionSettings", 4)?;
        s.serialize_field("enabled", &self.enabled)?;
        s.serialize_field("time_factor", &self.time_factor.unwrap_or(0.0))?;
        s.serialize_field("destroy_factor", &self.destroy_factor.unwrap_or(0.0))?;
        s.serialize_field("pollution_factor", &self.pollution_factor.unwrap_or(0.0))?;
        s.end()
    }
}
