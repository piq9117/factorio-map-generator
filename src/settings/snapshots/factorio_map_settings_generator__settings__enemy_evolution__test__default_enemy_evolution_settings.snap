---
source: src/settings/enemy_evolution.rs
expression: default_enemy_evolution
---
{
  "enabled": false,
  "time_factor": 0.0,
  "destroy_factor": 0.0,
  "pollution_factor": 0.0
}
